packages/@uppy/companion/src/server/helpers/request.js

```diff
@@ -0,0 +1,129 @@
+const http = require('http')
+const https = require('https')
+const dns = require('dns')
+const ipAddress = require('ip-address')
+
+function isIPAddress (address) {
+  const addressAsV6 = new ipAddress.Address6(address)
+  const addressAsV4 = new ipAddress.Address4(address)
+  return addressAsV6.isValid() || addressAsV4.isValid()
+}
+
+/**
+ * Determine if a IP address provided is a private one.
+ * Return TRUE if it's the case, FALSE otherwise.
+ * Excerpt from:
+ * https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html#case-2---application-can-send-requests-to-any-external-ip-address-or-domain-name
+ *
+ * @param {string} ipAddress the ip address to validate
+ * @returns {boolean}
+ */
+function isPrivateIP (ipAddress) {
+  let isPrivate = false
+  // Build the list of IP prefix for V4 and V6 addresses
+  const ipPrefix = []
+  // Add prefix for loopback addresses
+  ipPrefix.push('127.')
+  ipPrefix.push('0.')
+  // Add IP V4 prefix for private addresses
+  // See https://en.wikipedia.org/wiki/Private_network
+  ipPrefix.push('10.')
```

Comment:
I don't believe this to be correct, i think there are actual WAN addresses that start with 10 and the 10. range should be more discriminatory

Reply:
i just checked but couldn't find any such claims online so it seems i was wrong